### DISCUSSION TITLE
Ctrl + [ remapping neovim

### DISCUSSION THREAD

**Discussion by xander412** [neovim/neovim#29889] [Q&A] [answered]
I recently shifted to Neo vim. Previously I used to map Ctrl + [ to Ctrl + t such that when I use ctags forward and backward are close. This used to work in vim, but is not working in neovim. I know Ctrl + [ is standard esc in many terminals. But my point is that the mapping used to in vim.

**Comment by telemachus** [comment]
Can you please say more? How are you mapping the key. (I.e., what code in what file?) And how is it not working? What happens?

**Comment by xander412** [comment]
Previous code in .vimrc for vim : 
`nnoremap <C-[> <C-t>`

New binding created for neovim in init.lua (one which is outside of lua folder)
`vim.keymap.set('n', '<C-[>', '<C-t>', {silent = true, noremap = true})`

when I press C-[ which is mapped to C-t, vim should take me one level back in my ctags browse history. 
But this behaviour is not happening with the new binding. If I set the binding with some other key like C-x its working.

**Comment by zeertzjq** [comment]
What terminal emulator and keyboard layout are you using?

**Comment by xander412** [comment]
terminal emulator : konsole
layout - 'us'

**Reply by zeertzjq** [reply]
Hmm, that's strange, as I can get your mapping to work with the same terminal emulator and keyboard layout.

**Reply by xander412** [reply]
I am using nvChad, do you think it will have some effect?

**Reply by xander412** [reply]
++. When I using neovim --clean and set the mapping in normal mode, the mapping is working. But it is not working when I don't use --clean. Previously I also tried keymap delete before binding it in init.vim. But it was showing that no mapping was present before. Even if I do `:lua vim.keymap.set('n', '<C-[>', '<C-t>', {silent = true, noremap = true})` in neovim without --clean, its not working.

**Reply by zeertzjq** [reply]
You are using Konsole, which send the same sequence for `<C-[>` and `<Esc>`. Then it's possible that there is already a mapping for `<Esc>` present, in which case a mapping for `<C-[>` will have no effect in Konsole. Try inspecting the result of `:verbose nmap <C-[>` and `:verbose nmap <Esc>`. Also try mapping `<Esc>` instead.

**Reply by xander412** [reply]
Verbose prints : 

For :verbose nmap <C-[> 
```
n  <C-[>       * <C-T>                                                                                                                                                                                
        Last set from Lua
```


For :verbose nmap <Esc> 
```
n  <Esc>       * <Cmd> noh <CR>                                                                                                                                                                        
                 Clear highlights
        Last set from Lua
```

**Reply by zeertzjq** [reply]
Then you can try one of the following two:
1. Unmap `<Esc>`, then create your `<C-[>` mapping.
2. Use `<Esc>` instead of `<C-[>` as the lhs of your mapping.

**Reply by xander412** [reply, marked as answer]
Hi @zeertzjq  You are correct, there was an in-built binding for <Esc> in NvChad that was causing this.  Refer : [ NvChad Discussion ](https://github.com/NvChad/NvChad/discussions/2949)
